### PR TITLE
Sort search front-end

### PR DESCRIPTION
**Pull request by davidpmccormick**
References #1107

## Type
✨ Feature  

## Value
Adding front-end for radio/checkbox inputs and the sort search results form.

### Inputs
screen shot 2017-07-05 at 15 36 33

### Smaller screens
screen shot 2017-07-05 at 15 35 14

### Larger screens
screen shot 2017-07-05 at 15 35 27

### All
- [ ] Demoed to the relevant people
- [x] PR labelled and assigned
- [ ] Any introduced code complexity has been flagged

### UI component only
- [x] A11y tested: `npm run test:accessibility <URL>`
- [x] Browser tested: `npm run test:browsers <URL>`
- [x] Works without JS in the client
- [ ] Relevant tracking/monitoring has been considered

### CODE REVIEW

File: client/js/components/sort-search.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import showHide from './show-hide';
+import { nodeList } from '../util';
+
+const sortSearch = (el) => {
+  const toggle = showHide({
+    el,
+    activeClass: 'is-sort-search-active'
+  });
+
+  const form = el.querySelector('.js-sort-search-form');
```
Comment (jamesgorrie) [inline client/js/components/sort-search.js]: `fastDom`?

File: client/js/components/sort-search.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  const inputEls = nodeList(form.getElementsByTagName('input'));
+
+  inputEls.forEach((el) => {
+    el.addEventListener('change', () => {
+      form.submit();
+    });
+  });
+
+  toggle.trigger.addEventListener('click', toggle.toggleActive);
```
Comment (jamesgorrie) [inline client/js/components/sort-search.js]: What's this doing?

File: server/views/components/sort-search/sort-search.config.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      {
+        label: 'Oldest',
+        type: 'radio',
+        id: 'sort-oldest',
+        name: 'sort',
+        value: 'ASC'
+      }
+    ]
+  }
+};
```
Comment (jamesgorrie) [inline server/views/components/sort-search/sort-search.config.js]: Why not build these directly into the component?
I can't really see the benefit of this being config?

Comment (davidpmccormick) [inline server/views/components/sort-search/sort-search.config.js]: Fair point. Will update.

File: client/js/components/sort-search.js (hunk 2 above)
Comment (davidpmccormick) [inline client/js/components/sort-search.js]: it's from the `show-hide.js` component we have that toggles opening/closing of a component section while handling all the required ARIA attrs etc.

File: client/js/components/sort-search.js (hunk 1 above)
Comment (davidpmccormick) [inline client/js/components/sort-search.js]: Ah yeah. Will update/

File: client/js/components/sort-search.js (hunk 2 above)
Comment (jamesgorrie) [inline client/js/components/sort-search.js]: Ah.... Nice and stateful.

File: client/scss/utilities/_root-scope-classes.scss [hunk 4]
```diff
@@ -40,6 +40,20 @@
   justify-content: center;
 }
 
+.flex--h-space-between {
+  justify-content: space-between;
+}
+
+.plain-button {
```
Comment (jamesgorrie) [inline client/scss/utilities/_root-scope-classes.scss]: In the button stylesheet?
`btn--plain`?

Comment (davidpmccormick) [inline client/scss/utilities/_root-scope-classes.scss]: The problem with this as it stands is that the base `btn` class is working quite hard already (and as such `btn--plain` would require much more undoing of styles). Perhaps there's a piece of work to change `btn`'s default styling in another PR.

**Pull request merged**